src/backend/cuda/kernel/ireduce.hpp

```diff
@@ -57,13 +79,16 @@ namespace kernel
         __host__ __device__ MinMaxOp(T val, uint idx) :
             m_val(val), m_idx(idx)
         {
+            if (isNan(val)) {
+                m_val = Binary<T, af_max_t>::init();
+            }
         }
 
         __host__ __device__ void operator()(T val, uint idx)
         {
-            if (cabs(val) > cabs(m_val) ||
-                (cabs(val) == cabs(m_val) &&
-                 idx <= m_idx)) {
+            if (!isNan(val) &&
```

Comment:
Switch order of isNan operation. See other comments for info.

Reply:
👍